src/FromFile.jl

```diff
@@ -46,16 +86,40 @@ function from_m(m::Module, s::LineNumberNode, path::String, root_ex::Expr)
         file_module_sym = Symbol(relpath(path, pathof(root)))
     end
 
+    file_module = lazy_load_file(path, root, file_module_sym)
+    return load_symbols_from_file(file_module, file_module_sym, parse_import_stmts(root_ex))
+end
+
+function parse_import_stmts(root_ex::Expr)
+    import_exs = if root_ex.head === :block
+        filter(ex -> !(ex isa LineNumberNode), root_ex.args)
+    else
+        [root_ex]
+    end
+
+    all(ex -> ex.head === :using || ex.head === :import, import_exs) || error("expected using/import statement")
+    return import_exs
+end
+
+function lazy_load_file(path::String, root::Module, file_module_sym::Symbol)
+    return lazy_load_file(()->path, root, file_module_sym)
+end
```

Comment:
Pretty minor nit, but I wouldn't bother defining a new overload just for a single call site. Simpler to just do `()->path` where this is called.